File: Solutions/795. Number of Subarrays with Bounded Maximum/795. Number of Subarrays with Bounded Maximum.cpp
class Solution {
public:
    int numSubarrayBoundedMax(vector<int>& nums, int L, int R) {
        int res=0,prev=-1,dp=0;
        for(int i=0;i<nums.size();++i){
            if(nums[i]<L and i>0){
                res+=dp;
            }
            if(nums[i]>R){
                dp=0;
                prev=i;
            }
            if(L<=nums[i] and nums[i]<=R){
                dp=i-prev;
                res+=dp;
            }
        }
        return res;
    }
};

File: Solutions/795. Number of Subarrays with Bounded Maximum/795. Number of Subarrays with Bounded Maximum.py
class Solution:
    def numSubarrayBoundedMax(self, nums: List[int], L: int, R: int) -> int:
        res,prev=0,-1
        dp=0
        for i in range(len(nums)):
            if nums[i]<L and i>0:
                res+=dp
            if nums[i]>R:
                dp=0
                prev=i
            if L<=nums[i] and nums[i]<=R:
                dp=i-prev
                res+=dp
        return res